### PR TITLE
Add search to system profile /sap_sids endpoint

### PR DESCRIPTION
**Pull request by ryandillinfelton**
Hooks up inventory to the new xjoin-search feature allowing users to search for sap_sids with partial matching. Because it's reliant on xjoin-search to work I cannot create a full integration test, please test manually to check that it's working (it does on my machine)

### CODE REVIEW

File: api/system_profile.py [hunk 1]
```diff
@@ -135,6 +140,12 @@ def get_sap_sids(tags=None, page=None, per_page=None, staleness=None, registered
     if registered_with:
         variables["hostFilter"]["NOT"] = {"insights_id": {"eq": None}}
 
+    if search:
+        variables["filter"] = {
+            # Escaped so that the string literals are not interpretted as regex
```
Comment (kruai) [inline api/system_profile.py:145]: If you wanna keep this comment, I'd recommend changing "interpretted" to "interpreted"

**Review by kruai — APPROVED** [review]
There's one typo in a comment, but everything LGTM

**Review by thearifismail — APPROVED** [review]

File: api/system_profile.py (hunk 1 above)
Comment (ryandillinfelton) [reply api/system_profile.py:145]: Lol, copy pasted the comment from elsewhere in the code. I'll update both. Good eye! :eye:

**Comment by ryandillinfelton** [comment]
Fixed the typo Asa pointed out in my copy of the comment and the original in tag.py. No logic changes in the last commit

**Review by tahmidefaz — APPROVED** [review]
LGTM :+1:

**Pull request merged**